include/circt/Scheduling/Algorithms.h

```diff
@@ -74,6 +74,15 @@ LogicalResult scheduleLP(Problem &prob, Operation *lastOp);
 /// \p lastOp.
 LogicalResult scheduleLP(CyclicProblem &prob, Operation *lastOp);
 
+/// Solve the resource-constrained cyclic problem using constraint programming
+/// and an external SAT solver. The objectives are to determine the smallest
+/// feasible initiation interval, while not exceeding resource limits (defined
+/// by <tt> limit = k </tt> attributes in the \p operatortypes attribute
+/// dictionary), and to minimize the start time of the given \p lastOp.
+/// Fails if the dependence graph contains cycles that do not include at least
+/// one edge with a non-zero distance, or \p prob does not include \p lastOp.
```

Comment:
This describes the wrong problem :wink:

May I propose:
```suggestion
/// Solve the acyclic problem with shared operators using constraint programming
/// and an external SAT solver. The objective is to minimize the start time of
/// the given \p lastOp. Fails if the dependence graph contains cycles, or \p
/// prob does not include \p lastOp.
```

The formulation *does* compute a schedule with the optimal makespan, correct?